Mathlib/Tactic/ToAdditive/Frontend.lean

```diff
@@ -692,23 +692,40 @@ def reorderLambda (reorder : List (List Nat) := []) (src : Expr) : MetaM Expr :=
   lambdaTelescope src fun xs e => do
     mkLambdaFVars (xs.permute! reorder) e
 
+/-- Unfold `proof_nnn` aux lemmas using `unfoldAuxLemmas` in the type and value of `decl`.
+We have found that these kinds of aux lemmas sometimes do not have enough context to be additivized.
+TODO: consider trying to additivize them anyway, and on failure resort to unfolding them. -/
+def declUnfoldAuxLemmas (decl : ConstantInfo) : MetaM ConstantInfo := do
```

Comment:
Could you add a todo comment that the current behavior has bad performance and that this can probably be fixed if we were able to not unfold proofs generated by `declAbstractNestedProofs` (I don't know exactly how much instructions are spent on `to_additive`, but I think this is a significant increase)?

Reply:
Would you mind pushing the todo you want to see?

Reply:
Done